src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Packed.cs

```diff
@@ -0,0 +1,873 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+#pragma warning disable IDE0060 // https://github.com/dotnet/roslyn-analyzers/issues/6228
+
+#pragma warning disable 8500 // sizeof of managed types
+
+namespace System
+{
+    // This is a separate class instead of 'partial SpanHelpers' to hide the private helpers
+    // included in this file which are specific to the packed implementation.
+    internal static partial class PackedSpanHelpers
+    {
+        // We only do this optimization on X86 as the packing is noticeably more expensive on ARM in comparison.
```

Comment:
I see us commonly writing x86 to mean 32 bit, whereas you mean x86/x64. Not sure what the general term is, if we burned x86. Same on Arm, we sometimes use it to mean 32 bit only since I don't think Arm32 is a thing.

Reply:
Happy to change it to whatever the preferred language is.

Do you think a slight rewording like so would help clear it up?
```c#
// We only do this optimization if we have support for X86 intrinsics (Sse2) as the packing is noticeably cheaper compared to ARM (AdvSimd).
// While the impact on the worst-case (match at the start) is minimal on X86, it's prohibitively large on ARM.
```

Reply:
That reads better. @jkotas  do we have a standard term for x86/x64 - "x86 architecture"? Similar to "Arm architecture" to mean both bitnesses?

Reply:
I do not think there is a non-ambiguous term here. Explicitly mention both if you would like to avoid ambiguity. For example:

Our [documentation for `System.Runtime.Intrinsics.X86' namespace](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.x86) says "x86 and x64 systems".

Intel [documentation](https://www.intel.com/content/www/us/en/developer/articles/technical/intel-sdm.html) says "Intel® 64 and IA-32 architectures".